ApollosProject/apollos-prototype
# Allow During Onboarding the Skipping of Birthdate, Gender, or Both

## DESCRIPTION

### What does this PR do, or why is it needed?

Allows users to skip the birthdate, gender, or both fields when onboarding. I did this by slightly changing the form and mutation that manage these. The form now checks the values and dynamically builds the input to the mutation.

### What design trade-offs/decisions were made?

- The structure of the mutation on the client had to change since it was previously static
- Selecting "Prefer not to reply" will not actually save any data for gender, so it will be as if the user never input anything. This also means that changing from "male" or "female" to "prefer not to reply" will not actually change gender.

### How do I test this PR?

Best done on a new user

1. In onboarding with nothing selected, ensure that the user can skip
2. In onboarding, select "prefer not to reply"; no data should be saved
3. In onboarding, select a birthdate, but not a gender; you should advance and the birthdate should be saved.
4. In onboarding, keep your birthdate and select a gender; you should advance and your data should be saved

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
